spec/ruby/core/kernel/rand_spec.rb

```diff
@@ -52,6 +52,19 @@
   it "returns a float for an range argument where max is < 1" do
     rand(0.25..0.75).should be_kind_of(Float)
   end
+
+  it "returns a numeric for an range argument where max is < 1" do
+    require 'bigdecimal'
```

Comment:
@fengb your proposed patch adds conditionals for `is_a?(Integer)` and `respond_to?(:to_f)`. Those cases should be possible to cover with specs that neither need BigDecimal, nor an alternate Float implementation.

Am I missing something else?